docker/solr/entrypoint.sh

```diff
@@ -0,0 +1,213 @@
+            echo "SOLR_DOCS not defined, skipping initial indexing"
+        else
+            post_documents_solr5 $dir_name $SOLR_COLLECTION $SOLR_DOCS $SOLR_PORT
+        fi
+    else
+        add_core $dir_name $dir_conf $SOLR_CORE "$SOLR_CONFS"
+        run $dir_name $SOLR_PORT $SOLR_CORE
+         if [ -z "${SOLR_DOCS}" ]
+        then
+            echo "SOLR_DOCS not defined, skipping initial indexing"
+        else
+            post_documents $dir_name $SOLR_DOCS $SOLR_CORE $SOLR_PORT
+        fi
+    fi
+}
+
+add_core() {
+    dir_name=$1
+    dir_conf=$2
+    solr_core=$3
+    solr_confs=$4
+    # prepare our folders
+    [[ -d "${dir_name}/example/multicore/${solr_core}" ]] || mkdir $dir_name/example/multicore/$solr_core
+    [[ -d "${dir_name}/example/multicore/${solr_core}/conf" ]] || mkdir $dir_name/example/multicore/$solr_core/conf
+
+    # copy text configs from default single core conf to new core to have proper defaults
+    cp -R $dir_name/example/solr/conf/{lang,*.txt} $dir_name/example/multicore/$solr_core/conf/
+
+    # copies custom configurations
+    if [ -d "${solr_confs}" ] ; then
+      cp -R $solr_confs/* $dir_name/example/multicore/$solr_core/conf/
+      echo "Copied $solr_confs/* to solr conf directory."
+    else
+      for file in $solr_confs
+      do
+        if [ -f "${file}" ]; then
+            cp $file $dir_name/example/multicore/$solr_core/conf
+            echo "Copied $file into solr conf directory."
+        else
+            echo "${file} is not valid";
+            exit 1
+        fi
+      done
+    fi
+
+    # enable custom core
+    if [ "$solr_core" != "core0" -a "$solr_core" != "core1" ] ; then
+        echo "Adding $solr_core to solr.xml"
+        sed -i -e "s/<\/cores>/<core name=\"$solr_core\" instanceDir=\"$solr_core\" \/><\/cores>/" $dir_name/example/multicore/solr.xml
+    fi
+}
+
+post_documents() {
+    dir_name=$1
+    solr_docs=$2
+    solr_core=$3
+    solr_port=$4
+      # Post documents
+    if [ -z "${solr_docs}" ]
+    then
+        echo "SOLR_DOCS not defined, skipping initial indexing"
+    else
+        echo "Indexing $solr_docs"
+        java -Dtype=application/json -Durl=http://localhost:$solr_port/solr/$solr_core/update/json -jar $dir_name/example/exampledocs/post.jar $solr_docs
+    fi
+}
+
+create_collection() {
+    dir_name=$1
+    name=$2
+    dir_conf=$3
+    solr_port=$4
+    ./$dir_name/bin/solr create -c $name -d $dir_conf -shards 1 -replicationFactor 1 -p $solr_port
+    echo "Created collection $name"
+}
+
+post_documents_solr5() {
+    dir_name=$1
+    collection=$2
+    solr_docs=$3
+    solr_port=$4
+     # Post documents
+    if [ -z "${solr_docs}" ]
+    then
+        echo "SOLR_DOCS not defined, skipping initial indexing"
+    else
+        echo "Indexing $solr_docs"
+        echo "./$dir_name/bin/post -c $collection $solr_docs -p$solr_port"
+        ./$dir_name/bin/post -c $collection $solr_docs -p $solr_port
+    fi
+}
+
+download_and_run 3.6.0
```

Comment:
you don't need `download_and_run` anymore, as you download at build time now. You only need `run` here.